src/OrchardCore.Modules/OrchardCore.AdminDashboard/Controllers/DashboardController.cs

```diff
@@ -19,14 +19,13 @@
 namespace OrchardCore.AdminDashboard.Controllers
 {
     [Admin]
-    public class DashboardController : Controller
+    public class DashboardController : Controller, IUpdateModel
     {
         private readonly IAuthorizationService _authorizationService;
         private readonly IAdminDashboardService _adminDashboardService;
         private readonly IContentManager _contentManager;
         private readonly IContentItemDisplayManager _contentItemDisplayManager;
         private readonly IContentDefinitionManager _contentDefinitionManager;
-        private readonly IUpdateModelAccessor _updateModelAccessor;
```

Comment:
lolz, this was originally done to make it replacable (from memory), so suspect you are breaking something for someone here.

Reply:
If this was the case, it would be replaced in a test case. I don't think we are replacing it from a test so it should be fine.

Reply:
haha. you're dreaming if you think the tests cover everything :)

Reply:
That's not what I meant. We have a filter that changes the `modelBinderAccessor.ModelUpdater` used on the controller which reads it from the services collection. So if someone want to provider their own implementation of `IUpdateModelAccessor` the updated code will read their custom implementation and use. So it should not break anyone.

https://github.com/OrchardCMS/OrchardCore/blob/d99deaae6465b688eef259e7c65105fca3a9ea41/src/OrchardCore/OrchardCore.DisplayManagement/ModelBinding/ModelBinderAccessorFilter.cs#L19-L20

Reply:
This is where the story begins: https://github.com/OrchardCMS/OrchardCore/commit/110b8d71ee7c9b757abe1cafd6ae64e5bbcd2b08#diff-9a84d05c6fa5bd910a407bba4bb42b815a4753331061a6bcb06a5f97aed8432c In 2016 :D.

I think the issue here is that the current implementation allows you to swap out `IUpdateModel` between `ModelBinderAccessorFilter` and the actual model binding happening. With these changes, this wouldn't be possible. Now, I don't know where this can be useful, but it's a capability lost.

@sebastienros you have written this originally, any clues?

Reply:
Memory tells me it might have originally been @ns8482e that wanted this, but I don't know if the thing he originally wanteed it for ever got merged.

Reply:
We'll wait to hear from @sebastienros or @ns8482e. But if you are trying to swap implementation, you can do that by registering a different implementation in the service container and the same instance will be resolved.

Reply:
Where resolved, do you mean? In the example of this controller, `IContentItemDisplayManager` and the drivers down the line will use the instance of the controller as `IUpdateModel`. You could previously override that via `IUpdateModelAccessor.ModelUpdater` but not anymore.

Reply:
@Piedone look at the comment above https://github.com/OrchardCMS/OrchardCore/pull/15473#discussion_r1516873089

The same instance is already set at every controller from the filter.  So if you resolve that updater directly from the controller or you resolver it from the container, you'll get the same instance. At the end of the day, they all resolve the instance that you registered in the container.

Reply:
On line 75 the controller instance is passed to `_contentItemDisplayManager.BuildDisplayAsync()`. Due to this, it doesn't matter what's in `IUpdateModelAccessor.ModelUpdater`, display management will use the controller.

Before this change, the (current) controller being used as `IUpdateModel` was one option that was the default, set by the filter you mention. Now it's the only option, and even if you set ``IUpdateModelAccessor.ModelUpdater` to something else, still, the controller will be used. That's why I'm saying it's a capability lost (whether it's a useful capability is a different question, and I'm not sure).

Since almost all of the cases where the current `IUpdateModel` instance is accessed are in drivers, and thus coming similarly from controllers (there are a handful of cases of `IUpdateModelAccessor` being used in services and views), this PR effectively removes the ability to use anything else than the current controller for `IUpdateModel`.

Again, I'm not sure if this is a bad thing, but it's an impactful one.

Reply:
Looking at this more, there is no way to override the update logic that asp.net core. So the idea of changing the implementation is not ideal because even if you do, you can't change the updater behavior that is in the BasecController. I am wondering if we really need to care about allowing others to change the behavior of IUpdateModel since that will only change the logic partially "not for the methods found in the controller".

Reply:
@sebastienros thought here? Do we really need or should care about injecting `IUpdateModelAccessor` into controllers where there is no pure way to change the updater behavior in the app entirely even when we want to?